apm-lambda-extension/Makefile

```diff
@@ -60,7 +60,7 @@ publish: validate-layer-name validate-aws-default-region
 	@aws lambda \
 		--output json \
 		publish-layer-version \
-		--layer-name "${ELASTIC_LAYER_NAME}" \
+		--layer-name "$(ELASTIC_LAYER_NAME)-$(ARCHITECTURE)" \
```

Comment:
The make goal `publish` is the one in charge to apply the layer-name with the architecture since the Makefile is in charge to transform the GOARCH in the expected format.